### DISCUSSION TITLE
Error after make

### DISCUSSION THREAD

**Discussion by timofeyp** [strongswan/strongswan#1185] [Q&A] [answered]
Im trying to install strongswan 5.9.7 in ubuntu 22, but after run configure and make, I have an error.

`/configure --prefix=/usr --sysconfdir=/etc --enable-ha --enable-libipsec --enable-mysql --enable-monolithic  --enable-android --enable-attr-sql --enable-dhcp --enable-sql --enable-silent-rules      --enable-charon --enable-systemd --enable-ikev2 --enable-vici --enable-swanctl        --enable-nonce --enable-random --enable-drbg --enable-openssl --enable-curl           --enable-pem --enable-x509 --enable-constraints --enable-revocation --enable-pki      --enable-pubkey --enable-socket-default --enable-kernel-netlink --enable-resolve      --enable-eap-identity --enable-eap-md5 --enable-eap-dynamic --enable-eap-tls          --enable-updown --enable-tss-tss2 --enable-tpm  --enable-curl --enable-tss-tss2 --with-systemdsystemunitdir`
 
Result is here: [https://pastebin.com/LPLLfJHZ](https://pastebin.com/LPLLfJHZ)

Then `make`
 
And I have an error:
```
Making all in ipsec
Making all in charon
  CCLD     charon
/usr/bin/ld: /usr/lib/ipsec/libtls.so.0: undefined reference to `diffie_hellman_group_is_ec'
/usr/bin/ld: /usr/lib/ipsec/libtls.so.0: undefined reference to `prf_plus_create'
/usr/bin/ld: /usr/lib/ipsec/libtls.so.0: undefined reference to `diffie_hellman_group_names'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:538: charon] Error 1
make[2]: *** [Makefile:541: all-recursive] Error 1
make[1]: *** [Makefile:604: all-recursive] Error 1
make: *** [Makefile:515: all] Error 2
```
Full result of `make` is here: [https://pastebin.com/RjXm3nR1](https://pastebin.com/RjXm3nR1)

**Comment by strongX509** [comment, marked as answer]
Is /usr/lib/ipsec/libtls.so.0 a a leftover library from a previous strongSwan version? With strongSwan 5.9.7  we renamed diffie_hellman_group_t to key_exchange_method_t so there is no backwards compatibility to the binary ABI. Remove everything in /usr/lib/ipsec and /usr/libexec/ipsec before installing strongSwan. 5.9.7
